### PR TITLE
[ticket/14200] Allow hidden users to see himself on viewonline

### PR DESCRIPTION
**Pull request by Zoddo**
[PHPBB3-14200](https://tracker.phpbb.com/browse/PHPBB3-14200)

### CODE REVIEW

File: phpBB/includes/functions.php [hunk 1]
```diff
@@ -4299,7 +4299,7 @@ function obtain_users_online_string($online_users, $item_id = 0, $item = 'forum'
 					$row['username'] = '<em>' . $row['username'] . '</em>';
 				}
 
-				if (!isset($online_users['hidden_users'][$row['user_id']]) || $auth->acl_get('u_viewonline'))
+				if (!isset($online_users['hidden_users'][$row['user_id']]) || $auth->acl_get('u_viewonline') || $row['user_id'] == $user->data['user_id'])
```
Comment (CHItA) [inline phpBB/includes/functions.php]: I would prefer `===`

File: phpBB/viewonline.php [hunk 2]
```diff
@@ -187,7 +187,7 @@
 
 		if (!$row['session_viewonline'])
 		{
-			$view_online = ($auth->acl_get('u_viewonline')) ? true : false;
+			$view_online = ($auth->acl_get('u_viewonline') || $row['user_id'] == $user->data['user_id']) ? true : false;
```
Comment (CHItA) [inline phpBB/viewonline.php]: same here

**Comment by CHItA** [comment]
Thanks for the contribution :+1:

File: phpBB/includes/functions.php (hunk 1 above)
Comment (Zoddo) [inline phpBB/includes/functions.php]: Done

**Pull request merged**